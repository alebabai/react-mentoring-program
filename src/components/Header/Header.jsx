import React from 'react'

import Logo from '../Logo'
import Title from '../Title'
import Search from '../Search'
import SearchLink from '../SearchLink'
import Movie from '../Movie'


const DefaultHeaderContent = () => (
    <>
        <Logo value="netflixroulette" />
        <Title value="Find your movie" />
        <Search />
    </>
)

const MovieHeaderContent = props => (
    <>
        <SearchLink />
        <Movie {...props} />
    </>
)

export const Header = ({ movie }) => (
    <header>
        {movie ? <MovieHeaderContent {...movie} /> : <DefaultHeaderContent />}
    </header>
)

export default Header
