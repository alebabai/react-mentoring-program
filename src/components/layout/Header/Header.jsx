import React from 'react'

export const Header = ({ children }) => (
    <header>
        {children}
    </header>
)

export default Header
