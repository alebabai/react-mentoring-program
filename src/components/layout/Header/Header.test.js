import React from 'react'

import Header from './Header'

import renderer from 'react-test-renderer';

describe('Should match snapshot', () => {
    test('empty', () => {
        const component = renderer.create(
            <Header />
        )
        expect(component.toJSON()).toMatchSnapshot()
    })

    test('with children', () => {
        const component = renderer.create(
            <Header>
                <div className="wrapper-1">
                    <h1>Some title 1</h1>
                </div>
                <div className="wrapper-2">
                    <h1>Some title 2</h1>
                </div>
            </Header>
        )
        expect(component.toJSON()).toMatchSnapshot()
    })
})

