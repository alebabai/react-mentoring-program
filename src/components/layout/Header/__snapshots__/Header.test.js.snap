// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot empty 1`] = `<header />`;

exports[`Should match snapshot with children 1`] = `
<header>
  <div
    className="wrapper-1"
  >
    <h1>
      Some title 1
    </h1>
  </div>
  <div
    className="wrapper-2"
  >
    <h1>
      Some title 2
    </h1>
  </div>
</header>
`;
